Automattic/jetpack
# VideoPress dashboard: improve upload error handler

Fixes https://github.com/Automattic/videopress/issues/1038

## Proposed changes:
This PR addresses unhandled upload errors on VideoPress dashboard, providing error components for both list and row views of VideoPress dashboard.

### Other information:


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

The next instructions should be tested on:
- [ ] self hosted/JN
- [ ] simple sites (simple sites don't use VideoPress dashboard, but uploads should be tested on these as well)
- [ ] AT sites (should mostly behave as self hosted, since the standalone plugin takes place, but unsure how to get the latest code builds to AT)

Create an instance, follow the upgrade flow for VideoPress. Once done, you should be able to access VideoPress' dashboard at `wp-admin/admin.php?page=jetpack-videopress#/`

- Verify the video uploads work on dashboard
- Verify the video uploads work on the block editor
- Verify the video uploads work on the media library

The upload process should not show any improvements or handling, but it should work as before.

- Install "Disable XML-RPC" [plugin](https://wordpress.org/plugins/disable-xml-rpc-api/) and activate it

At this point, the XML backend connection should not be working, hence, video uploads should fail

Go back to VideoPress dashboard and upload a video. It should fail and the video card should show a warning sign. Click on the button to get some context and a link to our troubleshooting guide. Click on the trash icon to get rid of the error.

<img width="302" alt="image" src="https://github.com/user-attachments/assets/5f48c064-58b8-4037-aaf1-783941ea3c7b">
<img width="621" alt="image" src="https://github.com/user-attachments/assets/a6e63f46-8786-4b4a-9762-740aab88cbee">

Do it again, but switch to list/grid view and verify the same functionality exists.
<img width="830" alt="image" src="https://github.com/user-attachments/assets/401b17df-6711-4daf-93ee-1210841e9122">
<img width="841" alt="image" src="https://github.com/user-attachments/assets/4f75c8c9-a19b-4924-a856-f4e71fd949d6">


Disabling "Disable XML-RPC" plugin should get things back to normal, but beware of this step on local environments as it turns to get tricky (unsure why).
